libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxSpeechApi.kt

```
@@ -78,71 +75,49 @@ class MapboxSpeechApi @JvmOverloads constructor(
-                                available
-                            )
-                        )
-                    )
-                }
+                val fallback = getFallbackAnnouncement(voiceInstruction)
+                val speechError = SpeechError(result.exception, null, fallback)
+                consumer.accept(ExpectedFactory.createError(speechError))
             }
         }
     }
 
-    private suspend fun processVoiceAnnouncement(
-        voiceInstruction: VoiceInstructions,
-        onAvailable: (SpeechAnnouncement) -> Unit
-    ) {
-        val checkVoiceInstructionsResult =
-            VoiceProcessor.process(VoiceAction.PrepareTypeAndAnnouncement(voiceInstruction))
-        when (checkVoiceInstructionsResult as VoiceResult.VoiceTypeAndAnnouncement) {
-            is VoiceResult.VoiceTypeAndAnnouncement.Success -> {
-                val announcement = voiceInstruction.announcement()
-                val ssmlAnnouncement = voiceInstruction.ssmlAnnouncement()
-                // Can't be null as it's checked in processVoiceAnnouncement
-                val available = SpeechAnnouncement.Builder(announcement!!)
-                    .ssmlAnnouncement(ssmlAnnouncement)
-                    .build()
-                onAvailable(available)
-            }
-            is VoiceResult.VoiceTypeAndAnnouncement.Failure -> {
-                check(false) {
-                    "Invalid state: processVoiceAnnouncement can't produce " +
-                        "Failure VoiceTypeAndAnnouncement VoiceResult"
-                }
-            }
+    @Throws(IllegalStateException::class)
+    private fun getFallbackAnnouncement(voiceInstruction: VoiceInstructions): SpeechAnnouncement {
+        if (VoiceInstructionsParser.parse(voiceInstruction).isError) {
+            throw IllegalStateException(
+                "Invalid state: processVoiceAnnouncement can't produce " +
+                    "Fallback VoiceTypeAndAnnouncement VoiceResult"
```

Error message needs to be changed since there is no action, result and processor anymore in this PR.

---

Looks like `VoiceInstructionParser` returns the following `error` 
```
private fun invalidInstructionsError() =
        Error("VoiceInstructions ssmlAnnouncement / announcement can't be null or blank")
```
We can just propagate this error as `IllegalStateException`

---

Done.